xhr/send-redirect.htm

```diff
@@ -9,20 +9,28 @@
   <body>
     <div id="log"></div>
     <script>
-      function redirect(code) {
-        var test = async_test(document.title + " (" + code + ")")
+      function redirect(code, method="GET", redirectLocation="content.py") {
+        var test = async_test(`${document.title} (${code}, ${method}, ${redirectLocation})`)
         test.step(function() {
           var client = new XMLHttpRequest()
           client.onreadystatechange = function() {
             test.step(function() {
               if(client.readyState == 4) {
-                assert_equals(client.getResponseHeader("x-request-method"), "GET")
-                assert_equals(client.getResponseHeader("x-request-content-type"), "application/x-pony")
+                let expected_method = method;
+                if (method == "POST" && (code == "301" || code == "302"))
+                  expected_method = "GET";
+                if (code == "303" && method != "GET" && method != "HEAD")
+                  expected_method = "GET";
+                assert_equals(client.getResponseHeader("x-request-method"), expected_method)
+                let expected_content_type = "application/x-pony";
+                if (method == "POST" && (code == "301" || code == "302" || code == "303"))
```

Comment:
This is not equivalent to the two conditionals for expected_method above, yet in the specification that logic is shared.

I.e., while this only expects Content-Type to be dropped when the method is POST, the specification also expects it to be dropped when the method is CHICKEN (or some such).

Merging the code might be clearer.

(See step 12 of https://fetch.spec.whatwg.org/#concept-http-redirect-fetch.)